qiskit/cluster.py

```diff
@@ -7,30 +7,45 @@
 
 class Processor:
 
-    def __init__(self, cluster, index, nr_qubits):
+    def __init__(self, cluster, index, nr_qubits,flag):
         self.cluster = cluster
+        self.flag=flag
         self.qc = cluster.qc
         self.index = index
         names = ['alice', 'bob', 'charlie', 'david', 'eve', 'frank', 'george', 'harry']
         if index > len(names):
             self.name = str(index)
         else:
             self.name = names[index]
-        self.main_reg = QuantumRegister(nr_qubits, f'{self.name}_main')
-        self.qc.add_register(self.main_reg)
-        self.teleport_reg = QuantumRegister(1, f'{self.name}_teleport')
-        self.qc.add_register(self.teleport_reg)
-        self.entanglement_reg = QuantumRegister(1, f'{self.name}_entanglement')
-        self.qc.add_register(self.entanglement_reg)
-        self.measure_reg = ClassicalRegister(2, f'{self.name}_measure')
-        self.qc.add_register(self.measure_reg)
+
+        if flag == "teleport":
```

Comment:
An enum (https://docs.python.org/3/library/enum.html) might be cleaner than a string. If you use a string, you may want to throw an exception if it has an unexpected value.